src/list/SortedLinkedList.java

```
@@ -1,7 +1,91 @@
 package list;
 
-public class SortedLinkedList {
+public class SortedLinkedList<S extends Comparable<S>> implements SortedList<S> {
+
+    private int count = 0;
+
+    private Node head;
+
+    @Override
+    public void insert(S element) {
+        Node n = new Node(element, null);
+        Node pred = null;
+        Node succ = head;
+        while (succ != null && element.compareTo(succ.Element) > 0 ){
+            pred = succ;
+            succ = succ.Next;
+        }
+        if (pred == null) {
+            head = n;
+        }
+        else {
+            pred.Next = n;
+        }
+        n.Next = succ;
+        count++;
+    }
+
+    @Override
+    public boolean delete(int index) {
+        Node pred = null;
+        Node n = head;
+        int i = 0;
+        while(i < index && n != null){
+            pred = n;
+            n = n.Next;
+            i++;
+        }
+        if (n != null){
+            if(pred == null){
+                head = n.Next;
+            }else {
+                pred.Next = n.Next;
+            }
+            count--;
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public S get(int index) {
+        Node n = head;
+        int i = 0;
+        while(i < index && n != null){
```

Here you are checking implicitly the index out of bounds (`n != 0`), but an earlier and dedicated fail  is to be preferred.
(Fail fast)